IDR/notebook/scripts/nbconverted/1.describe_screens.py

```diff
@@ -0,0 +1,310 @@
+                stain,
+                stain_target,
+                pixel_size_x,
+                pixel_size_y
+            ])
+
+    # Output results
+    plate_results_df = pd.DataFrame(
+        plate_results,
+        columns=[
+            "screen_id",
+            "plate_id",
+            "plate_name",
+            "image_id",
+            "cell_line",
+            "strain",
+            "gene_identifier",
+            "phenotype_identifier",
+            "stain",
+            "stain_target",
+            "pixel_size_x",
+            "pixel_size_y"
+        ]
+    )
+    return plate_results_df
+
+
+# In[10]:
+
+
+# Load IDR ids
+data_dir = pathlib.Path(
+    "../data")
+id_file = pathlib.Path(data_dir, "idr_ids.tsv")
+id_df = pd.read_csv(id_file, sep="\t")
+
+
+# In[11]:
+
+
+# Create http session
+INDEX_PAGE = "https://idr.openmicroscopy.org/webclient/?experimenter=-1"
+with requests.Session() as session:
+    request = requests.Request('GET', INDEX_PAGE)
+    prepped = session.prepare_request(request)
+    response = session.send(prepped)
+    if response.status_code != 200:
+        response.raise_for_status()
+
+
+# In[12]:
+
+
+# Extract summary details for all screens
+screen_ids = id_df.query("category=='Screen'").id.tolist()
+screen_details_df = (
+    pd.concat([
+       extract_study_info(session=session, screen_id=x) for x in screen_ids
+    ], axis="rows")
+    .reset_index(drop=True)
+)
+
+output_file = pathlib.Path(data_dir, "screen_details.tsv")
+screen_details_df.to_csv(output_file, index=False, sep="\t")
+
+
+# In[13]:
+
+
+# Initialize Pool object for threading
+start = time.time()
+available_cores = len(os.sched_getaffinity(0))
+pool = multiprocessing.Pool(processes=available_cores)
+print(f"\nNow processing {len(screen_ids)} screens with {available_cores} cpu cores.\n")
+
+# Collect subset of metadata
+test_indices = [0, 1, 2]
+test_screen_ids = [screen_ids[screen] for screen in test_indices]
+
+# Pull pertinent details about the screen (plates, wells, channels, cell line, etc.)
+plate_results_dfs = pool.map(describe_screen, test_screen_ids)
+
+# Terminate pool processes
+pool.close()
+pool.join()
+
+
+# In[14]:
+
+
+# Combine to create full dataframe
+all_plate_results_df = pd.concat(plate_results_dfs, ignore_index=True)
+
+# Collect imaging method metadata for each screen
+img_screen_index = dict()
+for index in screen_details_df.itertuples(index=False):
+    screenID = index[19]
+    img_type = index[5]
+    img_screen_index[screenID] = img_type
+
+# Map imageing method per screen to the final data frame
+all_plate_results_df["imaging_method"] = all_plate_results_df["screen_id"].map(
+    img_screen_index)
+
+print(f'Metadata collected. Running cost is {(time.time()-start)/60:.1f} min. ', 'Now saving file.')
+
+# Save data frame as a single parquet file
+output_file = pathlib.Path(data_dir, "plate_details_per_screen.parquet.gzip")
+all_plate_results_df.to_parquet(output_file, compression='gzip')
```

Comment:
interesting! I know that Dave recommends `snappy` compression. No need to change anything here, but I am wondering if there is a difference.

Reply:
I'll look into it!